### PR TITLE
Support kind keyword in Opencensus

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/19364

### CODE REVIEW

File: sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py [hunk 1]
```diff
@@ -61,7 +61,21 @@ def __init__(self, span=None, name="span", **kwargs):
...
+            OpenCensusSpanKind.CLIENT if value == SpanKind.CONSUMER else # No consumer in opencensus
+            OpenCensusSpanKind.UNSPECIFIED if value == SpanKind.INTERNAL else # No internal in opencensus
+            OpenCensusSpanKind.UNSPECIFIED if value == SpanKind.UNSPECIFIED else
+            None
+        ) # type: SpanKind
+        if value and kind is None:
+            raise ValueError("Kind {} is not supported in OpenCensus".format(value))
         self._span_instance = span or tracer.start_span(name=name, **kwargs)
+        if kind is not None:
+            self._span_instance.span_kind = kind
```
Comment (xiangyan99) [inline sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py:78]: No handler for links?

Comment (xiangyan99) [reply sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py:78]: Or it is handled in tracer.start_span?

File: sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py [hunk 2]
```diff
@@ -61,7 +61,38 @@ def __init__(self, span=None, name="span", **kwargs):
...
+        ) # type: SpanKind
+        if value and kind is None:
+            raise ValueError("Kind {} is not supported in OpenCensus".format(value))
+
+        links = kwargs.pop('links', None)
         self._span_instance = span or tracer.start_span(name=name, **kwargs)
+        if kind is not None:
+            self._span_instance.span_kind = kind
+
+        if links:
```
Comment (xiangyan99) [inline sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py:82]: Missed links= kwargs.pop('links', None). :)

Comment (rakshith91) [reply sdk/core/azure-core-tracing-opencensus/azure/core/tracing/ext/opencensus_span/__init__.py:82]: i have it in line 77

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**